File: src/main/java/com/example/appjamback/domain/user/service/UserLoginService.java
package com.example.appjamback.domain.user.service;

import com.example.appjamback.domain.user.domain.User;
import com.example.appjamback.domain.user.exception.PasswordMissMatchException;
import com.example.appjamback.domain.user.facade.UserFacade;
import com.example.appjamback.domain.user.presentation.dto.request.LoginRequest;
import com.example.appjamback.global.security.dto.TokenResponse;
import com.example.appjamback.global.security.jwt.JwtTokenProvider;
import lombok.RequiredArgsConstructor;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@RequiredArgsConstructor
public class UserLoginService {
    private final UserFacade userFacade;
    private final JwtTokenProvider jwtProvider;
    private final PasswordEncoder passwordEncoder;

    @Transactional(readOnly = true)
    public TokenResponse execute(LoginRequest request) {

        User user = userFacade.getUserByAccountId(request.getAccountId());

        if (!passwordEncoder.matches(user.getPassword(),request.getPassword())) {
            throw PasswordMissMatchException.EXCEPTION;
        }
        return jwtProvider.getToken(user.getAccountId());
    }
}

File: src/main/kotlin/com/example/instargram/domain/user/service/UserLoginService.kt
package com.example.instargram.domain.user.service

import com.example.instargram.domain.user.exception.PasswordMissMatchException
import com.example.instargram.domain.user.exception.UserNotFoundException
import com.example.instargram.domain.user.facade.UserFacade
import com.example.instargram.domain.user.presentation.dto.request.LoginRequest
import com.example.instargram.global.security.jwt.JwtTokenProvider
import com.example.instargram.global.security.jwt.dto.TokenResponse
import org.springframework.security.crypto.password.PasswordEncoder
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class UserLoginService(
    private val passwordEncoder: PasswordEncoder,
    private val jwtTokenProvider: JwtTokenProvider,
    private val userFacade: UserFacade,
) {

    @Transactional(readOnly = true)
    fun execute(request: LoginRequest): TokenResponse {
        val user = userFacade.getUserByInfo(request.info)?: throw UserNotFoundException
        if (!passwordEncoder.matches(request.password, user.password)) throw PasswordMissMatchException
        return jwtTokenProvider.getToken(user.info)
    }
}
